src/core/public/core_system.test.ts

```diff
@@ -173,14 +181,29 @@ describe('#start()', () => {
     expect(mockInstance.start).toHaveBeenCalledTimes(1);
     expect(mockInstance.start).toHaveBeenCalledWith();
   });
+});
 
-  it('calls legacyPlatform#start()', () => {
-    startCore();
-    const [mockInstance] = MockLegacyPlatformService.mock.instances;
-    expect(mockInstance.start).toHaveBeenCalledTimes(1);
-    expect(mockInstance.start).toHaveBeenCalledWith({
-      injectedMetadata: mockInjectedMetadataStartContract,
-      fatalErrors: mockFatalErrorsStartContract,
+describe('LegacyPlatform targetDomElement', () => {
+  it('only mounts the element when started, before starting the legacyPlatformService', () => {
+    const rootDomElement = document.createElement('div');
+    const core = new CoreSystem({
+      ...defaultCoreSystemParams,
+      rootDomElement,
+    });
+
+    const [legacyPlatform] = MockLegacyPlatformService.mock.instances;
+
+    let targetDomElementParentInStart: HTMLElement;
+    (legacyPlatform as any).start.mockImplementation(() => {
+      targetDomElementParentInStart = targetDomElement.parentElement;
     });
+
+    // targetDomElement should not have a parent element when the LegacyPlatformService is constructed
+    const { targetDomElement } = MockLegacyPlatformService.mock.calls[0][0];
+    expect(targetDomElement).toHaveProperty('parentElement', null);
+
+    // starting the core system should mount the targetDomElement as a child of the rootDomElement
+    core.start();
+    expect(targetDomElementParentInStart!).toBe(rootDomElement);
```

Comment:
I was specifically trying to verify that the parent was set when `legacyPlatform.start()` was executed. If we only checked after calling `core.start()` then it could technically be added to the `rootDomElement` after calling `legacyPlatform.start()`.